parachain/pallets/outbound-queue/src/test.rs

```diff
@@ -266,3 +266,47 @@ fn encode_digest_item() {
 		);
 	});
 }
+
+#[test]
+fn validate_messages_with_fees() {
+	new_tester().execute_with(|| {
+		let message = mock_message(1000);
+		let (_, fee) = OutboundQueue::validate(&message).unwrap();
+		assert_eq!(fee.local, 698000000);
+		assert_eq!(fee.remote, 2680000000000);
+	});
+}
+
+#[test]
+fn test_calculate_fees() {
+	new_tester().execute_with(|| {
+		let gas_used: u64 = 250000;
+		let illegal_price_params: PricingParameters<<Test as Config>::Balance> =
+			PricingParameters {
+				exchange_rate: FixedU128::from_rational(1, 400),
+				fee_per_gas: 10000_u32.into(),
+				rewards: Rewards { local: 1_u32.into(), remote: 1_u32.into() },
+			};
+		let fee = OutboundQueue::calculate_fee(gas_used, illegal_price_params);
+		assert_eq!(fee.local, 698000000);
+		assert_eq!(fee.remote, 1000000);
+	});
+}
+
+#[test]
+fn test_calculate_fees_with_valid_exchange_rate_but_remote_fee_calculated_as_zero() {
+	new_tester().execute_with(|| {
+		let gas_used: u64 = 250000;
+		let illegal_price_params: PricingParameters<<Test as Config>::Balance> =
+			PricingParameters {
+				exchange_rate: FixedU128::from_rational(1, 1),
+				fee_per_gas: 1_u32.into(),
+				rewards: Rewards { local: 1_u32.into(), remote: 1_u32.into() },
+			};
+		let fee = OutboundQueue::calculate_fee(gas_used, illegal_price_params.clone());
+		assert_eq!(fee.local, 698000000);
+		// Though none zero pricing params the remote fee calculated here is invalid
+		// which should be avoided
+		assert_eq!(fee.remote, 0);
+	});
+}
```

Comment:
It seems odd that the calculated remote fee is zero. Can you investigate why this happens? Seems like it could be a bug in our code.

Reply:
Actually I realize why the calculated remote fee is zero. Since ether has 18 decimal places and, dot has 10 decimal places, `convert_from_ether_decimals` will end up truncating to zero very small fixed-point numbers that don't fit in 10 decimals.

This is fine though, and it is expected behavior.